sdk/translation/azure-ai-translation-document/tests/test_list_submitted_jobs.py

```diff
@@ -4,77 +4,304 @@
-            else:
-                self._validate_translation_job(job)
\ No newline at end of file
+            self._validate_translation_job(job)
+
+
+    @pytest.mark.skip(reason="filter not working!")
+    @DocumentTranslationPreparer()
+    @DocumentTranslationClientPreparer()
+    def test_list_submitted_jobs_filter_by_status(self, client):
+        jobs_count = 10
+        docs_per_job = 3
+
+        # create some jobs
+        job_ids = self._create_and_submit_sample_translation_jobs(client, jobs_count, wait=False, docs_per_job=docs_per_job)
+
+        # update status
+        cancelled_count = jobs_count//2
+        for id in job_ids[:cancelled_count]:
+            client.cancel_job(id)
+        statuses = ["Cancelled", "Cancelling"]
+        self.wait(10) # wait for cancelled to propagate
+
+        # list jobs
+        submitted_jobs = list(client.list_submitted_jobs(statuses=statuses))
+        self.assertIsNotNone(submitted_jobs)
+
+        # check statuses
+        for job in submitted_jobs:
+            self.assertIn(job.status, statuses)
+
+
+    @DocumentTranslationPreparer()
+    @DocumentTranslationClientPreparer()
+    def test_list_submitted_jobs_filter_by_ids(self, client):
+        jobs_count = 3
+        docs_per_job = 2
+
+        # create some jobs
+        job_ids = self._create_and_submit_sample_translation_jobs(client, jobs_count, wait=False, docs_per_job=docs_per_job)
+
+        # list jobs
+        submitted_jobs = list(client.list_submitted_jobs(ids=job_ids))
+        self.assertIsNotNone(submitted_jobs)
+
+        # check statuses
+        for job in submitted_jobs:
+            self.assertIn(job.id, job_ids)
+
+
+    @DocumentTranslationPreparer()
+    @DocumentTranslationClientPreparer()
+    def test_list_submitted_jobs_filter_by_created_after(self, client):
+        # create some jobs
+        jobs_count = 3
+        docs_per_job = 2
+
+        # create some jobs
+        start = datetime.now()
+        job_ids = self._create_and_submit_sample_translation_jobs(client, jobs_count, wait=False, docs_per_job=docs_per_job)
+
+        # list jobs
+        submitted_jobs = list(client.list_submitted_jobs(created_after=start))
+        self.assertIsNotNone(submitted_jobs)
+
+        # check statuses
+        for job in submitted_jobs:
+            self.assertIn(job.id, job_ids)
+            assert(job.created_on.replace(tzinfo=None) >= start.replace(tzinfo=None))
+
+
+    @pytest.mark.skip(reason="for some reason, jobs created after 'end' timestamp showup in result! might be different timestamps locally and service")
+    @DocumentTranslationPreparer()
+    @DocumentTranslationClientPreparer()
+    def test_list_submitted_jobs_filter_by_created_before(self, client):
+        '''
+            NOTE: maybe we need to wait for few seconds after calling 'end = datetime.now()'
+            for the local and service clocks to differ by some significant amount 
```

Comment:
Did the waiting help or do you still see it fail on occasion?

Reply:
just figured out that time filters are timezone aware!